File: src/main/java/com/teste/entrevista/staff/model/Atendimento.java
package com.teste.entrevista.staff.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Enumerated;
import jakarta.persistence.EnumType;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import java.time.LocalTime;
import lombok.Data;

@Entity
@Data
public class Atendimento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "veiculo_id")
    private Carro veiculo;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    private LocalDate dataAgendamento;
    private LocalTime horarioAgendamento;

    @Enumerated(EnumType.STRING)
    private TipoAtendimento tipo;

    @Enumerated(EnumType.STRING)
    private StatusAtendimento status;

    public enum TipoAtendimento {
        REPARO, MANUTENCAO_PREVENTIVA, LIMPEZA
    }

    public enum StatusAtendimento {
        AGENDADO, EM_ANDAMENTO, CONCLUIDO
    }
}

File: laravel-api/realtime-api/app/Models/Atendimento.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Atendimento extends Model {
    use HasFactory;

    protected $fillable = [
        'carro_id',
        'cliente_id',
        'data_agendamento',
        'horario_agendamento',
        'tipo',
        'status',
    ];

    public function carro() {
        return $this->belongsTo(Carro::class);
    }

    public function cliente() {
        return $this->belongsTo(Cliente::class);
    }
}
